bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -46,23 +65,59 @@
     private HashMap<String, Integer> typeTable = new HashMap<>();
     protected HashMap<String, String> stateNames = new HashMap<>();
 
+    protected String url = "";
+
+    private Map<String, Unit<?>> units = new HashMap<>();
+
     public SomfyTahomaBaseThingHandler(Thing thing) {
         super(thing);
+        // Define default units
+        units.put("Number:Temperature", SIUnits.CELSIUS);
+        units.put("Number:Energy", Units.WATT_HOUR);
+        units.put("Number:Illuminance", Units.LUX);
+        units.put("Number:Dimensionless", Units.PERCENT);
     }
 
     public HashMap<String, String> getStateNames() {
         return stateNames;
     }
 
-    protected String url = "";
-
     @Override
     public void initialize() {
-        url = getURL();
-        if (getThing().getProperties().containsKey(RSSI_LEVEL_STATE)) {
-            createRSSIChannel();
+        logger.debug("initializing thing handler for thing {}", getThing().getUID());
+        Bridge bridge = getBridge();
+        initializeThing(bridge != null ? bridge.getStatus() : null);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {} for thing {}", bridgeStatusInfo, getThing().getUID());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Reply:
I changed into `logger.trace`.
This is a log just to see when `bridgeStatusChanged` is called.

Reply:
You could use the debugger to see when the method is called. Here is an excerpt from the coding guidelines:

```
Logging is not a replacement for using the debugger.

void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

Reply:
But these bindiongs DEBUG logs like many others are logs that should help us, developers, to understand what's wrong with the binding when a user has a problem. We ask the user to enable the binding DEBUG logs and we can have more information.
If by debugger, you mean Eclipse debugger, of course I can use it myself but a standard user will never do it.

I will remove these logs but I do not understand the new philosophy with logs.

Reply:
Redundant logging doesn't seem right, just to avoid setting an additional log level. If the state of the Thing is important for tracking down a problem, the logging for the Thing state should be enabled.

Reply:
This is largely easier when you don't have to know and enable multiple logs, including logs from the core framework. Even a binding developer is not necessarily aware of how is implemented the core framework and what packages the logs have to be enabled. Even for the binding developer, it will require to search in the code of the core framework !
IMHO, reducing logs at the binding level will just add more difficulty to analyze problems reported by users.

Reply:
Log removed

Reply:
I think it's a documentation issue. I filed a pull-request openhab/openhab-docs#1510.